api/src/common/notification_helpers.py

```diff
@@ -30,6 +32,62 @@ def create_tagged_user_notification(
         send_push_notifications(notification)
 
 
+# A function that sends push notifications to a user when they recieve a connection request
+def create_connection_request_notification(
+    reciever: CustomUser,
+    sender: CustomUser,
+    is_request: bool,  # request => true if requesting, false if accepting
+) -> None:
+    content_type = ContentType.objects.get_for_model(sender)
+    content = f"{sender.preferred_name} sent you a connection request"
+    if is_request:
+        content = f"{sender.preferred_name} accepted your connection request"
+
+    notification = Notification(
+        content=content,
+        content_type=content_type,
+        object_id=sender.id,
+        creator_id=sender.id,
+    )
+    notification.save()
+    notification.receivers.add(reciever)
+    send_push_notifications(notification)
+
+
+# A function that sends push notifications to a user when they recieve a message
+def create_message_notification(
+    recievers: List[CustomUser], sender: CustomUser, message: Message
```

Comment:
Instead of List[CustomUser], we can do QuerySet[CustomUser]. This way, we don't need to convert to a list before passing in receivers to this method.

Also, typo in recieve (should be receive)